### PR TITLE
Adding more tests for methods in Utils::VersionControl::GitAnnex and converting over some methods to use ExtendedGit.

### PR DESCRIPTION
**Pull request by cgalarza**
- Adding more tests for `ExtendedGit` methods.
- Adding more tests for `Utils::VersionControl::GitAnnex`
- Replacing more shelled out git-annex commands with `ExtendedGit` calls
- Attempting to concentrate code that depends on being in a certain directory. Changing the directory leads to unintended consequences in other parts of the code.
- Added classes to `ExtendedGit` that parse output provided by git-annex commands. This addition allows for clearer tests.
- Deleted unused methods. 

In a future PR, I plan to refactor the `Utils::VersionControl::GitAnnex` to work slightly different. I can provide more details if that would be helpful. That probably won't be until more extensive testing of manifest loads.

### CODE REVIEW

File: spec/utils/version_control/git_annex_spec.rb [hunk 1]
```diff
@@ -38,4 +62,231 @@
...
+      before { git_annex.drop({ content: filename }, cloned_repo_path) }
+
+      it 'only drops README.md' do
+        expect(git.annex.whereis[filename].here?).to be false # README.md dropped.
+        expect(git.annex.whereis[new_file].here?).to be true # Other files NOT dropped.
+      end
+    end
+
+    context 'when imaging user configured' do
+      it 'changes permissions'
```
Comment (mksndz) [inline spec/utils/version_control/git_annex_spec.rb:192]: does this pass or fail? maybe you need `xit` here

Comment (cgalarza) [reply spec/utils/version_control/git_annex_spec.rb:192]: It's just listed as pending in the output. I could add a more descriptive pending message.

File: lib/extended_git/annex.rb [hunk 2]
```diff
@@ -5,35 +5,71 @@ def initialize(base)
...
+
+    def lock(path = nil)
+      lib.lock(path)
+    end
+
+    def whereis(path = nil, **options)
+      WhereIs.new(@base, path, options)
+    end
+
+    def info(options = {})
```
Comment (mksndz) [inline lib/extended_git/annex.rb:67]: looks like you don't need this param

Comment (cgalarza) [reply lib/extended_git/annex.rb:67]: 👍 Will remove!

File: lib/extended_git/annex_lib.rb [hunk 3]
```diff
@@ -75,29 +92,76 @@ def testremote(name, opts = {})
...
+      def escape(value)
+        value.nil? ? nil : value.shellescape
+      end
+
+      # Run a git annex command. This method takes a lot of inspiration from
+      # the `command` method in `Git::Lib`.
+      def command(cmd, opts = [])
+        # TODO:global config, pointing at correct working path and git dir
+        global_opts = []
+        global_opts << "--git-dir=#{@git_dir}" if !@git_dir.nil?
```
Comment (mksndz) [inline lib/extended_git/annex_lib.rb:147]:
```suggestion
        global_opts << "--git-dir=#{@git_dir}" if @git_dir
```

File: lib/extended_git/annex_lib.rb [hunk 4]
```diff
@@ -75,29 +92,76 @@ def testremote(name, opts = {})
...
+        global_opts << "--work-tree=#{@git_work_dir}" if !@git_work_dir.nil?
```
Comment (mksndz) [inline lib/extended_git/annex_lib.rb:148]:
```suggestion
        global_opts << "--work-tree=#{@git_work_dir}" if @git_work_dir
```

File: lib/extended_git/where_is.rb [hunk 5]
```diff
@@ -0,0 +1,66 @@
+module ExtendedGit
+  class WhereIs
```
Comment (mksndz) [inline lib/extended_git/where_is.rb:2]: It's hard from me to tell just from the included specs how this class is intended for use. Some top level documentation might be helpful.

File: spec/extended_git/annex_spec.rb [hunk 6]
```diff
@@ -0,0 +1,299 @@
...
+
+    it 'retrieves file from remote' do
+      expect { git.annex.get(readme) }.not_to raise_error
+      expect(File.exist?(File.join(git.dir.path, readme))).to be true
+    end
+  end
+
+  describe '#enableremote' do
+    include_context 'add readme to repository'
+    include_context 'add directory special remote'
```
Comment (mksndz) [inline spec/extended_git/annex_spec.rb:122]: this is great - a simple way to dry up specs without having to add a bunch of helper files :rocket:

**Review by mksndz — APPROVED** [review]

File: lib/extended_git/where_is.rb (hunk 5 above)
Comment (cgalarza) [reply lib/extended_git/where_is.rb:18]: I added some more documentation, but I think I should add more. I'll try to do it in a later PR.

Comment (mksndz) [reply lib/extended_git/where_is.rb:18]: very nice!

**Pull request merged**